types/graphql-upload/index.d.ts

```diff
@@ -47,4 +47,7 @@ export interface FileUpload {
   createReadStream(): ReadStream;
 }
 
-export type Upload = Promise<FileUpload>;
+export abstract class Upload {
```

Comment:
Is there any reason in particular that you made this an "abstract" class? It's not intended to be "subclassed" as it is both [declared](https://github.com/jaydenseric/graphql-upload/blob/0cbc9d90a181ace59be279fee4c3b269c47d1cbe/public/Upload.js#L30) and [directly instantiated](https://github.com/jaydenseric/graphql-upload/blob/0cbc9d90a181ace59be279fee4c3b269c47d1cbe/public/processRequest.js#L203).

```suggestion
export class Upload {
```

Reply:
Mostly because of how the the documentation

For example the [Upload class](https://www.npmjs.com/package/graphql-upload#class-upload)
is litterd with:
> This should only be utilized by processRequest.

While [processRequest](https://www.npmjs.com/package/graphql-upload#function-processrequest)
is very clear in that it:
> parses the operations and map fields to create an Upload instance for each expected file upload, placing references wherever the file is expected in the GraphQL operation for the Upload scalar to derive it’s value
and that is
> can be used to create custom middleware.

I just get the strong sens that it was never intended for the user to instantiate.  
Now i of course I don't **know** that that is the case. It just felt correct.  
Though it does feel unorthodox, and maybe DefinetlyTyped shouldn't care about the docs and simply type the JS. In any case if you feel its better to remove the ˋabstractˋ i have no objections.